src/bootstrap/src/core/builder.rs

```diff
@@ -439,7 +439,9 @@ impl StepDescription {
         }
 
         // strip CurDir prefix if present
-        let mut paths: Vec<_> = paths.iter().map(|p| p.strip_prefix(".").unwrap_or(p)).collect();
+        let curdir_path = PathBuf::from(".").canonicalize().unwrap();
+        let mut paths: Vec<_> =
+            paths.iter().map(|p| p.strip_prefix(&curdir_path).unwrap_or(p)).collect();
```

Comment:
I realized shortly after writing this that this change will break support for relative paths starting with `./`

We need to canonicalize both paths before stripping the prefix.